mvtprovider/postgis/README.md

```diff
@@ -0,0 +1,127 @@
+
+```toml
+[[mvt_providers.layers]]
+name = "landuse"
+# this table uses "geom" for the geometry_fieldname and "gid" for the id_fieldname so they don't need to be configured
+sql = "SELECT ST_AsMVTGeom(geom,!BBOX!) AS geom, gid FROM gis.landuse WHERE geom && !BBOX!"
+```
+
+### Provider Layers Properties
+
+- `name` (string): [Required] the name of the layer. This is used to reference this layer from map layers.
+- `geometry_fieldname` (string): [Optional] the name of the filed which contains the geometry for the feature. defaults to `geom`.
+- `id_fieldname` (string): [Optional] the name of the feature id field. defaults to `gid`.
+- `srid` (int): [Optional] the SRID of the layer. Supports `3857` (WebMercator) or `4326` (WGS84).
+- `geometry_type` (string): [Optional] the layer geometry type. If not set, the table will be inspected at startup to try and infer the gemetry type. Valid values are: `Point`, `LineString`, `Polygon`, `MultiPoint`, `MultiLineString`, `MultiPolygon`, `GeometryCollection`.
+- `sql` (string): [Required] custom SQL to use use. Supports the following tokens:
+  - `!BBOX!` - [Required] will be replaced with the bounding box of the tile before the query is sent to the database. `!bbox!` and`!BOX!` are supported as well for compatibilitiy with queries from Mapnik and MapServer styles.
+  - `!X!` - [Optional] will replaced with the "X" value of the requested tile.
+  - `!Y!` - [Optional] will replaced with the "Y" value of the requested tile.
+  - `!Z!` - [Optional] will replaced with the "Z" value of the requested tile.
+  - `!ZOOM!` - [Optional] will be replaced with the "Z" (zoom) value of the requested tile.
+  - `!SCALE_DENOMINATOR!` - [Optional] scale denominator, assuming 90.7 DPI (i.e. 0.28mm pixel size)
+  - `!PIXEL_WIDTH!` - [Optional] the pixel width in meters, assuming 256x256 tiles
+  - `!PIXEL_HEIGHT!` - [Optional] the pixel height in meters, assuming 256x256 tiles
+  - `!ID_FIELD!` - [Optional] the id field name
+  - `!GEOM_FIELD!` - [Optional] the geom field name
+  - `!GEOM_TYPE!` - [Optional] the geom type if defined otherwise ""
+
+## Example mvt_provider and map config
+
+When referencing the provider in the map section, one must prepend `mvt_` to the name. 
+
+Example:
+
+```toml
+[[mvt_providers]]
+name = "test_postgis"       
+type = "postgis"            
+host = "localhost"          
+port = 5432                 
+database = "tegola"         
+user = "tegola"             
+password = ""
+
+  [[mvt_providers.layers]]
+  name = "landuse"
+  sql = "SELECT ST_AsMVTGeom(geom,!BBOX!) AS geom, gid FROM gis.landuse WHERE geom && !BBOX!"
+
+[[maps]]
+name = "cities"
+center = [-90.2,38.6,3.0]  # where to center of the map (lon, lat, zoom)
+
+    [[maps.layers]]
+    name = "landuse"
+    # note the mvt_ prefix on the name of the provider.
+    provider_layer = "mvt_test_postgis.landuse" 
```

Comment:
I still think this is a confusing gotcha for the end-user. Ideally we could have the `type` use the `mvt_` prefix, but that does complicate the checking in another way. Just something we can think about from a UX perspective.

Reply:
#688 covers this.